### PR TITLE
feat: allow changing check interval

### PR DESCRIPTION
**Pull request by GenixPL**
- [feature] Added option to change check interval without creating a new instance.
- [fix] Corrected check interval's default value doc.

### CODE REVIEW

**Review by GenixPL — COMMENTED** [review]
i didnt wanna add new dependency ([fake_async](https://pub.dev/packages/fake_async)), or play on marginal time intervals, so the testing time has been extended by ~1.5s; can change if needed

**Comment by GenixPL** [comment]
i have found a problem, working on a fix

**Review by GenixPL — COMMENTED** [review]
- the problem ive mentioned was in the project ive used the package (ive created a dumb mock spamming in my tests)... 🤦🏻‍♂️;
- ive noticed ive run it with wrong dart line length formatting, so that's corrected;
- ive fixed the tests (sry for submitting the first one without running the full suite, just mine); i needed the test binding, so ive moved my tests to a separate file (the binding was causing the real life test to fail);
- ive changed the name of the "setter" to better indicate that the timer will be reset (i was considering ignoring the set/reset if the value was the same as current, but ended up leaving it; it doesnt seem critical in any way - both ways seem reasonable)

sry again for submitting the pr before running full test suite

File: lib/src/internet_connection.dart [hunk 1]
```diff
@@ -142,6 +146,20 @@ class InternetConnection {
...
+  /// Allows to change existing [_checkInterval].
+  ///
+  /// Resets the [_timerHandle] in the process.
+  void setIntervalAndResetTimer(Duration duration) {
+    _checkInterval = duration;
+    _timerHandle?.cancel();
+    _timerHandle = Timer(_checkInterval, _maybeEmitStatusUpdate);
+  }
+
+  /// Returns current [_checkInterval].
```
Comment (OutdatedGuy) [inline lib/src/internet_connection.dart:158]: Please update comment to `Returns the current duration between connection checks.`

File: lib/src/internet_connection.dart [hunk 2]
```diff
@@ -142,6 +146,20 @@ class InternetConnection {
...
+  Duration get checkInterval {
+    return _checkInterval;
+  }
```
Comment (OutdatedGuy) [inline lib/src/internet_connection.dart:161]: Please convert this to expression body / function.

File: test/internet_connection_interval_change_test.dart
Comment (OutdatedGuy) [inline test/internet_connection_interval_change_test.dart:1]: You can use the `setUp` function inside the `checkInterval` group to apply the Flutter binding for tests only in that group, like:

```dart
group('checkInterval', () {
  setUp(() {
      TestWidgetsFlutterBinding.ensureInitialized();
  });

  test('abc', () {});

  test('xyz', () {});
  
  // blah blah blah
});
```

**Review by OutdatedGuy — CHANGES_REQUESTED** [review]
Looks good, just some minor nit-picks.

**Review by OutdatedGuy — APPROVED** [review]
LGTM!

**Pull request merged**